pkg/terraform/workspace_test.go

```diff
@@ -0,0 +1,369 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package terraform
+
+import (
+	"bytes"
+	"context"
+	"testing"
+
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/test"
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+	k8sExec "k8s.io/utils/exec"
+	testingexec "k8s.io/utils/exec/testing"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	"github.com/crossplane-contrib/terrajet/pkg/resource/json"
+	tferrors "github.com/crossplane-contrib/terrajet/pkg/terraform/errors"
+)
+
+var (
+	lineage               = "very-cool-lineage"
+	terraformVersion      = "1.0.10"
+	version               = 1
+	serial                = 3
+	directory             = "./testdata"
+	changeSummaryAdd      = `{"@level":"info","@message":"Plan: 1 to add, 0 to change, 0 to destroy.","@module":"terraform.ui","@timestamp":"0000-00-00T00:00:00.000000+03:00","changes":{"add":1,"change":0,"remove":0,"operation":"plan"},"type":"change_summary"}`
+	changeSummaryUpdate   = `{"@level":"info","@message":"Plan: 0 to add, 1 to change, 0 to destroy.","@module":"terraform.ui","@timestamp":"0000-00-00T00:00:00.000000+03:00","changes":{"add":0,"change":1,"remove":0,"operation":"plan"},"type":"change_summary"}`
+	changeSummaryNoAction = `{"@level":"info","@message":"Plan: 0 to add, 0 to change, 0 to destroy.","@module":"terraform.ui","@timestamp":"0000-00-00T00:00:00.000000+03:00","changes":{"add":0,"change":0,"remove":0,"operation":"plan"},"type":"change_summary"}`
+)
+
+func newLogger() logging.Logger {
+	zl := zap.New(zap.UseDevMode(true))
+	return logging.NewLogrLogger(zl.WithName("provider-jet-test"))
+}
+
+func TestWorkspace_Apply(t *testing.T) {
```

Comment:
Nit: Underscores aren't particularly idiomatic in go function names in my experience, so I'd prefer us to avoid them in test function names too. I think `TestWorkspaceApply` would be fine here, for example.

https://github.com/crossplane/crossplane/blob/master/CONTRIBUTING.md#prefer-table-driven-tests

Reply:
Thank you @negz Resolved!